### PR TITLE
[ADD] auth_dynamic_groups

### PR DESCRIPTION
**Pull request by hbrunn**

### CODE REVIEW

File: auth_dynamic_groups/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+Usage
+-----
+Check `Dynamic` on a group you want to be dynamic. Now fill in the condition,
+using `user` which is a browse record of the user in question that evaluates
+truthy if the user is supposed to be a member of the group and falsy if not.
+
+There is a constraint on the field to check for validity if this expression.
+When you're satisfied, click the button `Evaluate` to prefill the group's
+members. The condition will be checked now for every user who logs in.
```
Comment (pedrobaeza) [inline auth_dynamic_groups/README.md]: An example of an expression would be good.

**Comment by pedrobaeza** [comment]
A little note. The rest seems excellent.

File: auth_dynamic_groups/README.md (hunk 1 above)
Comment (hbrunn) [inline auth_dynamic_groups/README.md]: thanks Pedro, see below

**Comment by pedrobaeza** [comment]
Cool, one last thing: please rename README.md to README.rst, because Odoo parsed language is reStructuredText (although it takes also .md files), and GitHub is going to be confused about the extension.

**Comment by hbrunn** [comment]
thanks, done

**Comment by pedrobaeza** [comment]
:+1:

File: auth_dynamic_groups/model/res_users.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        groups_obj = pool.get('res.groups')
+        user.write(
+            {
+                'groups_id': [
+                    (4, dynamic_group.id)
+                    if dynamic_group.eval_dynamic_group_condition()
+                    else (3, dynamic_group.id)
+                    for dynamic_group in groups_obj.browse(
+                        cr, uid,
+                        groups_obj.search(cr, uid,
```
Comment (StefanRijnhart) [inline auth_dynamic_groups/model/res_users.py]: Search and browse with SUPERUSER_ID to prevent problems with users who are not employees.

Comment (hbrunn) [inline auth_dynamic_groups/model/res_users.py]: done

**Comment by StefanRijnhart** [comment]
Thanks, tested succesfully. :+1:

**Pull request merged**